File: ios-app/CalendarioCiclismo/Views/Components/CCCard.swift
import SwiftUI

/// Tarjeta canónica de la app — superficie pulida única extraída del cintillo
/// "Hoy" (`TodayHighlightsBanner`). Centraliza el "tratamiento de tarjeta" para
/// que toda la UI iOS comparta el mismo lenguaje visual:
///
///  - Esquinas redondeadas continuas (18pt por defecto, igual que el cintillo).
///  - Superficie de material translúcido (`.regularMaterial`), como el cintillo,
///    en vez de un color plano — da profundidad sutil sobre el fondo.
///  - Tinte de marca opcional muy leve sobre el material (7% por defecto).
///  - Hairline de borde (`Color.primary.opacity(0.06)`, 0.5pt) para definir la
///    tarjeta sobre fondos claros, donde el material casi se funde con
///    `systemBackground`.
///  - Sombra suave (negro 6%, radio 8, y=3) idéntica a la del cintillo.
///
/// El contenido recibe el área interna ya recortada a la forma de la tarjeta;
/// cualquier gesto/ripple del contenido queda confinado a las esquinas.
///
/// Paridad de referencia: `TodayHighlightsBanner.bannerCard` / Android `CCCard`.
struct CCCard<Content: View>: View {
    /// Color de marca; si no es nulo, tiñe el material a `accentAlpha`.
    var accent: Color? = nil
    var accentAlpha: Double = 0.07
    var cornerRadius: CGFloat = 18
    /// Si es `false`, omite la sombra (útil para tarjetas embebidas en listas
    /// largas donde N sombras apiladas pesarían demasiado).
    var showShadow: Bool = true
    @ViewBuilder var content: () -> Content

    var body: some View {
        let shape = RoundedRectangle(cornerRadius: cornerRadius, style: .continuous)
        content()
            .background(cardSurface)
            .clipShape(shape)
            .overlay(
                // Hairline de definición sobre fondos claros.
                shape.strokeBorder(Color.primary.opacity(0.06), lineWidth: 0.5)
            )
            .shadow(
                color: showShadow ? Color.black.opacity(0.06) : .clear,
                radius: showShadow ? 8 : 0,
                x: 0,
                y: showShadow ? 3 : 0
            )
    }

    /// Superficie de la tarjeta: material translúcido con un tinte de marca muy
    /// leve encima cuando se proporciona `accent`.
    @ViewBuilder
    private var cardSurface: some View {
        ZStack {
            Rectangle().fill(.regularMaterial)
            if let accent {
                accent.opacity(accentAlpha)
            }
        }
    }
}

// MARK: - Modificador de superficie para tarjetas de detalle

extension View {
    /// Aplica el tratamiento de tarjeta canónico (esquinas + hairline + sombra
    /// suave) sobre un color de fondo PLANO — la variante "neutra fría" del
    /// rediseño, para secciones de detalle y Ajustes que son bloques de una
    /// sola entidad (no listados de carreras distintas). Sustituye al par
    /// `.background(AppTheme.cardBackground).clipShape(RoundedRectangle(...))`,
    /// añadiéndole el hairline y la sombra del cintillo "Hoy".
    ///
    /// Paridad: Android `CCCard` neutro (sin `accent`).
    func ccCardSurface(
        cornerRadius: CGFloat = 12,
        fill: Color = AppTheme.cardBackground,
        showShadow: Bool = true
    ) -> some View {
        let shape = RoundedRectangle(cornerRadius: cornerRadius, style: .continuous)
        return self
            .background(fill)
            .clipShape(shape)
            .overlay(
                shape.strokeBorder(Color.primary.opacity(0.06), lineWidth: 0.5)
            )
            .shadow(
                color: showShadow ? Color.black.opacity(0.06) : .clear,
                radius: showShadow ? 8 : 0,
                x: 0,
                y: showShadow ? 3 : 0
            )
    }
}

File: android-app/app/src/main/java/app/calendariociclismo/android/ui/components/CCCard.kt
package app.calendariociclismo.android.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.ElevatedCard
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/**
 * Tarjeta canónica de la app — superficie pulida única extraída del cintillo
 * "Hoy" (TodayHighlightsBanner). Centraliza el "tratamiento de tarjeta" para
 * que toda la UI Android comparta el mismo lenguaje visual:
 *
 *  - Esquinas redondeadas (18dp por defecto, igual que el cintillo).
 *  - Elevación tonal Material 3 (2dp) en vez del blur de iOS.
 *  - Tinte de marca opcional muy leve sobre el contenedor (7% por defecto).
 *  - Hairline de borde para definir la tarjeta sobre fondos claros, donde el
 *    contenedor elevado casi se funde con el `background`.
 *
 * El contenido recibe el área interna ya recortada a la forma de la tarjeta;
 * cualquier `clickable`/ripple del contenido queda confinado a las esquinas.
 *
 * Paridad de referencia: `TodayHighlightsBanner.BannerCarousel`.
 */
@Composable
fun CCCard(
    modifier: Modifier = Modifier,
    /** Color de marca; si no es nulo, tiñe el contenedor a [accentAlpha]. */
    accent: Color? = null,
    accentAlpha: Float = 0.07f,
    cornerRadius: Int = 18,
    elevation: Int = 2,
    content: @Composable () -> Unit,
) {
    val shape = RoundedCornerShape(cornerRadius.dp)
    // Contenedor explícito según el modo:
    //  - Neutro (sin accent): gris frío `surfaceVariant`, fijado a mano para no
    //    depender del tinte de elevación de M3 (que reintroducía el rosado).
    //  - Listados (con accent): superficie blanca `surface` + velo de marca en
    //    el Box interior, conservando el aspecto previo de Hoy.
    val containerColor =
        if (accent != null) MaterialTheme.colorScheme.surface
        else MaterialTheme.colorScheme.surfaceVariant
    ElevatedCard(
        shape = shape,
        colors = CardDefaults.elevatedCardColors(containerColor = containerColor),
        elevation = CardDefaults.elevatedCardElevation(defaultElevation = elevation.dp),
        modifier = modifier,
    ) {
        Box(
            modifier = Modifier
                .clip(shape)
                // Tinte de marca muy leve sobre el contenedor de la tarjeta.
                .then(
                    if (accent != null) Modifier.background(accent.copy(alpha = accentAlpha))
                    else Modifier
                )
                // Hairline de definición sobre fondos claros.
                .border(
                    width = 0.5.dp,
                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.06f),
                    shape = shape,
                ),
        ) {
            content()
        }
    }
}
